File: backend/mobile-leetnote-backend/src/main/java/com/example/leetnote_backend/model/DTO/ProblemListDTO.java
package com.example.leetnote_backend.model.DTO;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class ProblemListDTO {
    Long problemId;
    String title;
    String difficulty;
    boolean isFavorite;
    boolean isSolved;
}

File: frontend/mobile-leetnode-frontend/app/src/main/java/com/example/leetnote/data/model/ProblemListDTO.kt
package com.example.leetnote.data.model

import com.google.gson.annotations.SerializedName

data class ProblemListDTO (
    @SerializedName("problemId") val id: Long,
    val title: String,
    val difficulty: String,
    @SerializedName("solved")val isSolved: Boolean,
    @SerializedName("favorite")val isFavorite: Boolean
)

data class LeetProblem(
    val id: Long,
    val title: String,
    val difficulty: String,
    val isSolved: Boolean,
    val isFavorite: Boolean
)